libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -586,16 +587,17 @@ void ContractLevelChecker::checkPayableFallbackWithoutReceive(ContractDefinition
 
 void ContractLevelChecker::checkStorageSize(ContractDefinition const& _contract)
 {
-	bigint size = 0;
-	for (ContractDefinition const* contract: _contract.annotation().linearizedBaseContracts | ranges::views::reverse)
-		for (VariableDeclaration const* variable: contract->stateVariables())
-			if (!(variable->isConstant() || variable->immutable()))
-			{
-				size += variable->annotation().type->storageSizeUpperBound();
-				if (size >= bigint(1) << 256)
-				{
-					m_errorReporter.typeError(7676_error, _contract.location(), "Contract requires too much storage.");
-					break;
-				}
-			}
+	for (VariableDeclaration::Location location: {VariableDeclaration::Location::Unspecified, VariableDeclaration::Location::Transient})
+	{
+		std::optional<bigint> size = contractStorageSizeUpperBound(_contract, location);
+		if (!size)
+		{
+			if (location == VariableDeclaration::Location::Unspecified)
```

Comment:
This looks like a good opportunity to try out the `using enum` thingy from C++20:

```suggestion
	using enum VariableDeclaration::Location;
	for (VariableDeclaration::Location location: {Unspecified, Transient})
	{
		std::optional<bigint> size = contractStorageSizeUpperBound(_contract, location);
		if (!size)
		{
			if (location == Unspecified)
```

Reply:
Nice! This was just what I was wishing for here, I didn't know C++20 had it!